clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -546,6 +543,26 @@ public ConsumerRecords<K, V> poll(long timeout) {
         throw new KafkaException("method not implemented");
     }
 
+    <T> T tryGetFutureResult(
+            final Time time,
+            final WakeupableFuture<T> future,
+            final Duration timeout) throws ExecutionException, InterruptedException {
+        Timer timer = time.timer(timeout.toMillis());
+        do {
+            if (future.isDone()) {
+                return future.get();
+            }
+
+            if (this.shouldWakeup.get()) {
+                this.shouldWakeup.set(false);
+                future.cancel(true);
```

Comment:
Should `WakeupableFuture#wakeup` be used here?

Reply:
you are right!